ui/app/styles/components/actions.scss

```diff
@@ -117,14 +117,30 @@
       }
 
       .messages {
+        width: 100%;
         overflow: hidden;
 
-        code > pre {
-          height: 200px;
+        code {
           background-color: #0a0a0a;
           color: whitesmoke;
+          display: block;
+          overflow: auto;
+          height: 200px;
           border-radius: 6px;
           resize: vertical;
+          pre {
+            background-color: transparent;
+            color: unset;
+            overflow-anchor: none;
+            min-height: 100%;
+            white-space: pre-wrap;
+          }
```

Comment:
Part of this was making the scrollbars appear in the correct spot. An earlier attempt had me creating a scrollable area within the `<pre>` block, but that was unnecessary. Instead, we move the colo(u)r rules to the parent `<code>` element and have the `<pre>` always min-height to the code block's height.